tests/e2e/scenarios/test_reborn_webui_v2_legacy_tool_permissions.py

```
@@ -0,0 +1,457 @@
+            locked_id = locked["key"].removeprefix("tool.")
+            rejected = await client.post(
+                f"{reborn_v2_server}/api/webchat/v2/settings/tools/{locked_id}",
+                json={"state": "always_allow"},
+                timeout=15,
+            )
+            assert rejected.status_code >= 400
+
+
+async def test_reborn_legacy_always_approve_survives_reborn_restart(
+    reborn_v2_restartable_server,
+    reborn_approval_artifact_cleanup,
```

Ruff (0.15.18)

[warning] Unused function argument: `reborn_approval_artifact_cleanup`

(ARG001)